src/OrchardCore/OrchardCore/Modules/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -115,23 +115,38 @@ private static void AddExtensionServices(OrchardCoreBuilder builder)
         /// </summary>
         private static void AddStaticFiles(OrchardCoreBuilder builder)
         {
-            builder.Configure((app, routes, serviceProvider) =>
+            builder.ConfigureServices(services =>
             {
-                var env = serviceProvider.GetRequiredService<IHostingEnvironment>();
-                var appContext = serviceProvider.GetRequiredService<IApplicationContext>();
+                services.AddSingleton<IModuleStaticFileProvider>(serviceProvider => {
+                    var env = serviceProvider.GetRequiredService<IHostingEnvironment>();
+                    var appContext = serviceProvider.GetRequiredService<IApplicationContext>();
 
-                IFileProvider fileProvider;
-                if (env.IsDevelopment())
-                {
-                    var fileProviders = new List<IFileProvider>();
-                    fileProviders.Add(new ModuleProjectStaticFileProvider(appContext));
-                    fileProviders.Add(new ModuleEmbeddedStaticFileProvider(appContext));
-                    fileProvider = new CompositeFileProvider(fileProviders);
-                }
-                else
+                    IModuleStaticFileProvider fileProvider;
+                    if (env.IsDevelopment())
+                    {
+                        var fileProviders = new List<IFileProvider>
+                        {
+                            new ModuleProjectStaticFileProvider(appContext),
+                            new ModuleEmbeddedStaticFileProvider(appContext)
+                        };
+                        fileProvider = new ModuleCompositeFileProvider(fileProviders);
```

Comment:
So, here a `CompositeStaticFileProvider` defined in the same place as for `IStaticFileProvider` and `IVirtualPathBaseProvider`.